js2-mode.el

```diff
@@ -2632,14 +2632,17 @@ so many of its properties will be nil.
                                                         object
                                                         in-pos
                                                         each-pos
-                                                        foreach-p forof-p
+                                                        await-pos
+                                                        foreach-p forof-p forawait-p
                                                         lp rp)))
   "AST node for a for..in loop."
   iterator  ; [var] foo in ...
   object    ; object over which we're iterating
   in-pos    ; buffer position of 'in' keyword
   each-pos  ; buffer position of 'each' keyword, if foreach-p
+  await-pos  ; buffer position of 'await' keyword, if forawait-p
   foreach-p ; t if it's a for-each loop
```

Comment:
We could also set `foreach-p` to a symbol (e.g. `AWAIT`), instead of adding a new field. But I'm okay with either.

Reply:
I did it this way to not possibly break existing integrations outside this package.  If we're okay with removing backward compatibility we can use as symbol as you say.

Maybe in the future they will allow `for each await ()` ? :D

Reply:
Integrations that use an `(eq ... t)` check? I wouldn't be too worried about those.

Would `for each await` actually make any kind of sense?

Reply:
Rather even `(if foreach-p ...)` which would now be set to a symbol.  But of course we would rename the field so I guess in fact that mitigates the problem and should allow an easy migration path.

I'm honestly not even sure what `for each` does.  So probably not, never used it in my life.

Reply:
Actually if we don't rename the field, it should make the "migration" easier: only dependents that know about "for await" would recognize it, and the rest would continue to compile and treat the AST nodes as "for each". Which is the outdated syntax.

It was Mozilla's extension. The modern counterpart is "for of": https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Statements/for_each...in

Reply:
Again, I'm fine with either (rename of not). But reusing the field seems more economical.